homeassistant/components/homekit_controller/cover.py

```diff
@@ -265,30 +263,37 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
     async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
         """Move the cover tilt to a specific position."""
         tilt_position = kwargs[ATTR_TILT_POSITION]
+
+        if self.is_vertical_tilt:
+            char = self.service[CharacteristicsTypes.VERTICAL_TILT_CURRENT]
+        elif self.is_horizontal_tilt:
+            char = self.service[CharacteristicsTypes.HORIZONTAL_TILT_CURRENT]
+
+        # Calculate tilt_position. Convert from 1-100 scale to arc degree scale respecting possible min/max Values.
+        min_value = char.minValue
+        max_value = char.maxValue
+        if min_value is None or max_value is None:
+            raise ValueError(
+                "Entity does not provide minValue and maxValue for the tilt"
+            )
+
+        # inverted scale
+        if min_value == -90 and max_value == 0:
+            tilt_position = int(
+                tilt_position / 100 * (min_value - max_value) + max_value
+            )
+        else:
+            tilt_position = int(
+                tilt_position / 100 * (max_value - min_value) + min_value
+            )
+
         if self.is_vertical_tilt:
-            # Recalculate to convert from percentage scale to arcdegree scale.
-            scale = 0.9
-            if (
-                self.service[CharacteristicsTypes.VERTICAL_TILT_TARGET].minValue == -90
-                and self.service[CharacteristicsTypes.VERTICAL_TILT_TARGET].maxValue
-                == 0
-            ):
-                scale = -0.9
-            tilt_position = int(tilt_position * scale)
+            # Set tilt_position.
             await self.async_put_characteristics(
                 {CharacteristicsTypes.VERTICAL_TILT_TARGET: tilt_position}
             )
         elif self.is_horizontal_tilt:
-            # Recalculate to convert from percentage scale to arcdegree scale.
-            scale = 0.9
-            if (
-                self.service[CharacteristicsTypes.HORIZONTAL_TILT_TARGET].minValue
-                == -90
-                and self.service[CharacteristicsTypes.HORIZONTAL_TILT_TARGET].maxValue
-                == 0
-            ):
-                scale = -0.9
-            tilt_position = int(tilt_position * scale)
+            # Set tilt_position.
             await self.async_put_characteristics(
                 {CharacteristicsTypes.HORIZONTAL_TILT_TARGET: tilt_position}
```

Comment:
The last bit of what I was trying to explain is - You might only need a single async_put_characteristics here because char.type contains the uuid, which would get rid of the 2nd if. This is the bit I wasn't entirely sure of - char.type is still typed as a string and not CharacteristicsTypes. It might be weekly types enough not to matter.

Reply:
ok. Finally got your point. :)
Pushed.

Reply:
Shit I think it does not work ... need to re-run tests on my end ... hold on ...

Reply:
I had to revert. Was failing ... tried 2-3 different things ...
FYI - This is my first Python PR ... usually coding Scala, C# ..

Reply:
You wouldn't know :) let me try it on this end. We've got 2 weeks before beta is cut anyway.

Reply:
Thanks for giving it a try ... else I think what we did on this PR is already great improvement